langchain/llms/base.py

```diff
@@ -62,7 +90,11 @@ def generate(
             else:
                 missing_prompts.append(prompt)
                 missing_prompt_idxs.append(i)
+        self.callback_manager.on_llm_start(
+            {"name": self.__class__.__name__}, missing_prompts
+        )
         new_results = self._generate(missing_prompts, stop=stop)
+        self.callback_manager.on_llm_end(new_results)
         for i, result in enumerate(new_results.generations):
```

Comment:
A question, should we be logging one LLM call irrespective of the number of generations produced? Or one per generation

Reply:
hm thats a good question. i dont have conviction in either answer atm...

Reply:
yeah, let's maybe keep it as is right now and change if users feel strongly